tests/integration/characters/test_bob_handles_frags.py

```diff
@@ -206,13 +206,13 @@ def test_bob_can_issue_a_work_order_to_a_specific_ursula(enacted_federated_polic
 
     # Ursula decrypts the encrypted KFrag
     encrypted_kfrag = enacted_federated_policy.treasure_map.destinations[ursula.checksum_address]
-    alice = Alice.from_public_keys(verifying_key=enacted_federated_policy.alice_verifying_key)
+    alice = Alice.from_public_keys(verifying_key=enacted_federated_policy.publisher_verifying_key)
     plaintext_kfrag_payload = ursula.verify_from(stranger=alice,
                                                  message_kit=encrypted_kfrag,
                                                  decrypt=True)
     _signed_writ, the_kfrag = work_order.kfrag_payload_splitter(plaintext_kfrag_payload)
 
-    verified_kfrag = the_kfrag.verify(enacted_federated_policy.alice_verifying_key.as_umbral_pubkey())
+    verified_kfrag = the_kfrag.verify(enacted_federated_policy.publisher_verifying_key)
```

Comment:
How come the verification of a kfrag requires the `publisher_verifying_key`, as opposed to an `alice_verifying_key`? I understand that for some cases they can be one in the same, but if they are not, we should be able to obtain the `alice_verifying_key` instead.

Reply:
Agreed - this seems like the one situation when it's actually meant to be the kfrag generator's key, right?

Reply:
Yes, you are right - it was essentially a no-context fix, but looking at a slightly bigger picture, `enacted_federated_policy` should not have been used here at all (same goes for the `alice` above), and the key from `federated_alice` should have been used instead.

Reply:
Actually, in this whole bit, `AuthorizedKeyFrag` should have been used - fixed that.